contracts/EtomicSwapV2.sol

```diff
@@ -0,0 +1,476 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract EtomicSwapV2 {
+    using SafeERC20 for IERC20;
+
+    enum MakerPaymentState {
+        Uninitialized,
+        PaymentSent,
+        TakerSpent,
+        MakerRefunded
+    }
+
+    struct MakerPayment {
+        bytes20 paymentHash;
+        uint32 paymentLockTime;
+        MakerPaymentState state;
+    }
+
+    event MakerPaymentSent(bytes32 id);
+    event MakerPaymentSpent(bytes32 id, bytes32 secret);
+    event MakerPaymentRefundedTimelock(bytes32 id);
+    event MakerPaymentRefundedSecret(bytes32 id);
+
+    mapping(bytes32 => MakerPayment) public makerPayments;
+
+    enum TakerPaymentState {
+        Uninitialized,
+        PaymentSent,
+        TakerApproved,
+        MakerSpent,
+        TakerRefunded
+    }
+
+    struct TakerPayment {
+        bytes20 paymentHash;
+        uint32 immediateRefundTime;
+        uint32 paymentLockTime;
+        TakerPaymentState state;
+    }
+
+    event TakerPaymentSent(bytes32 id);
+    event TakerPaymentApproved(bytes32 id);
+    event TakerPaymentSpent(bytes32 id, bytes32 secret);
+    event TakerPaymentRefundedSecret(bytes32 id, bytes32 secret);
```

Comment:
Q: you removed takerSecret from `event MakerPaymentRefundedSecret`, but left it in `event TakerPaymentRefundedSecret`. Is it in purpose? or you just missed it.

Reply:
Related https://github.com/KomodoPlatform/etomic-swap/pull/6#discussion_r1499900383, we only need it for taker refund so that the maker can use it to refund their payments.